### ISSUE TITLE
Indepedent Gaussian mixture CV error

### ISSUE THREAD

**Issue by ramborz** [kimmo1019/Roundtrip#5]
I am running codes using TF 1.31.1 Python 3.7 venv. when I try to run example (1), when the Cross Validation starts I got runtime issues.
I used following command for faster run time debugging.
```sh
CUDA_VISIBLE_DEVICES=0 python main_density_est.py  --dx 2 --dy 2 --train True --data indep_gmm --epochs 10  --cv_epoch 3 --patience 5
```
When the CV starts at epoch 3 the following runtime warning shows
```sh
/home/connor/Roundtrip/venv/lib/python3.7/site-packages/numpy/core/fromnumeric.py:3420: RuntimeWarning: Mean of empty slice.  out=out, **kwargs)
/home/connor/Roundtrip/venv/lib/python3.7/site-packages/numpy/core/_methods.py:188: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
/home/connor/Roundtrip/venv/lib/python3.7/site-packages/numpy/core/_methods.py:262: RuntimeWarning: Degrees of freedom <= 0 for slice
  keepdims=keepdims, where=where)
/home/connor/Roundtrip/venv/lib/python3.7/site-packages/numpy/core/_methods.py:222: RuntimeWarning: invalid value encountered in true_divide
  subok=False)
/home/connor/Roundtrip/venv/lib/python3.7/site-packages/numpy/core/_methods.py:253: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
```
However the program keep running until it reaches last 2 IF conditions in function "train"  it will have local variable referenced before assignment. 
Any suggestions or solutions?
Thank you!

**Comment by kimmo1019** [comment]
> I am running codes using TF 1.31.1 Python 3.7 venv. when I try to run example (1), when the Cross Validation starts I got runtime issues.
> I used following command for faster run time debugging.
> 
> ```shell
> CUDA_VISIBLE_DEVICES=0 python main_density_est.py  --dx 2 --dy 2 --train True --data indep_gmm --epochs 10  --cv_epoch 3 --patience 5
> ```
> 
> When the CV starts at epoch 3 the following runtime warning shows
> 
> ```shell
> /home/connor/Roundtrip/venv/lib/python3.7/site-packages/numpy/core/fromnumeric.py:3420: RuntimeWarning: Mean of empty slice.  out=out, **kwargs)
> /home/connor/Roundtrip/venv/lib/python3.7/site-packages/numpy/core/_methods.py:188: RuntimeWarning: invalid value encountered in double_scalars
>   ret = ret.dtype.type(ret / rcount)
> /home/connor/Roundtrip/venv/lib/python3.7/site-packages/numpy/core/_methods.py:262: RuntimeWarning: Degrees of freedom <= 0 for slice
>   keepdims=keepdims, where=where)
> /home/connor/Roundtrip/venv/lib/python3.7/site-packages/numpy/core/_methods.py:222: RuntimeWarning: invalid value encountered in true_divide
>   subok=False)
> /home/connor/Roundtrip/venv/lib/python3.7/site-packages/numpy/core/_methods.py:253: RuntimeWarning: invalid value encountered in double_scalars
>   ret = ret.dtype.type(ret / rcount)
> ```
> 
> However the program keep running until it reaches last 2 IF conditions in function "train" it will have local variable referenced before assignment.
> Any suggestions or solutions?
> Thank you!

Neven met this error before. Roundtrip was developed in Python2.7 and Tensorflow 1.13.1.
Have you tried with python2?

**Comment by ramborz** [comment]
solved with conda python 2.7 Thankyou
